portal.lua

```diff
@@ -27,7 +27,7 @@ local function search(pos)
     local inZ = false
     for x = -1, 1 do
         if x ~= 0 then
-            local pos = { x = pos.x + x, y = pos.y, z = pos.z }
+            pos = { x = pos.x + x, y = pos.y, z = pos.z }
             if check(pos) then
                 table.insert(ptable, pos)
```

Comment:
This will overwrite the original `pos` and break the next search loops.
```suggestion
            local posX = { x = pos.x + x, y = pos.y, z = pos.z }
            if check(posX) then
                table.insert(ptable, posX)
```